### PR TITLE
build: update snappy to 1.1.5 and use its new CMake build system

### PR DESCRIPTION
**Pull request by tamird**
This updates to upstream's 1.1.5 plus one manual change: https://github.com/cockroachdb/snappy/commit/41e12f946060fa2d8fd105a85be81393ea4c6627.

### CODE REVIEW

File: build/teamcity-verify-archive.sh [hunk 1]
```diff
@@ -10,4 +10,4 @@ docker run \
   --rm \
   --volume="$(cd "$(dirname "$0")" && pwd):/work" \
   --workdir="/work" \
-  golang:1.8.1 ./verify-archive.sh
+  golang:stretch ./verify-archive.sh
```
Comment (benesch) [inline build/teamcity-verify-archive.sh]: This will become Go 1.9 the day Go 1.9 is released. I guess that's okay, and maybe even desirable?

**Review by benesch — APPROVED** [review]
LGTM, but I'll feel more comfortable if @petermattis agrees this is safe. Snappy release notes seem to indicate that 1.1.4 and 1.1.5 introduced only performance optimizations over 1.1.3, which is 👍.

File: build/teamcity-verify-archive.sh (hunk 1 above)
Comment (tamird) [inline build/teamcity-verify-archive.sh]: Yeah, I think that's fine.

**Comment by petermattis** [comment]
We're mid-cycle. Bumping the snappy version is fine by me.

**Pull request merged**